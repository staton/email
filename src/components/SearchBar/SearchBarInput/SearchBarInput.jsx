import React, { Component } from 'react';
import PropTypes from 'prop-types';
import STRINGS from '../../../resources/strings';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

const propTypes = {
    onBlur: PropTypes.func,
    onChange: PropTypes.func,
    onFocus: PropTypes.func,
    onKeyDown: PropTypes.func,
    placeholder: PropTypes.string
};

const defaultProps = {
    onBlur: () => {},
    onChange: () => {},
    onFocus: () => {},
    onKeyDown: () => {},
    placeholder: STRINGS.SearchPlaceholder
};

export class SearchBarInput extends Component {

    constructor() {
        super();
    }

	render() {
		return (
            <input 
                className="SearchBarInput"
                onBlur={this.props.onBlur}
                onChange={this.props.onChange}
                onFocus={this.props.onFocus}
                onKeyDown={this.props.onKeyDown}
                placeholder={this.props.placeholder}
                type="text"
                value={this.props.searchText} 
            />
		);
    }

}

function mapStateToProps(store, ownProps) {
    return {
        searchText: store.app.searchText,
        onBlur: ownProps.onBlur,
        onChange: ownProps.onChange,
        onFocus: ownProps.onFocus,
        onKeyDown: ownProps.onKeyDown
    };
}

function mapDispatchToProps(dispatch) {
    return bindActionCreators({
    },
    dispatch);
}

SearchBarInput.propTypes = propTypes;
SearchBarInput.defaultProps = defaultProps;

export default connect(mapStateToProps, mapDispatchToProps)(SearchBarInput);