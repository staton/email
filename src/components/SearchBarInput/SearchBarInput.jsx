import React, { Component } from 'react';
import PropTypes from 'prop-types';
import STRINGS from '../../resources/strings';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

const propTypes = {
    onBlur: PropTypes.func,
    onChange: PropTypes.func,
    onFocus: PropTypes.func,
    onKeyPress: PropTypes.func,
    placeholder: PropTypes.string
};

const defaultProps = {
    onBlur: () => {},
    onChange: () => {},
    onFocus: () => {},
    onKeyPress: () => {},
    placeholder: STRINGS.SearchPlaceholder
};

export class SearchBarInput extends Component {

    constructor() {
        super();
    }

	render() {
		return (
            <input 
                className="SearchBarInput"
                onBlur={this.props.onBlur}
                onChange={this.props.onChange}
                onFocus={this.props.onFocus}
                onKeyPress={this.props.onKeyPress}
                placeholder={this.props.placeholder}
                type="text"
                value={this.props.searchText} 
            />
		);
    }

}

function mapStateToProps(store, ownProps) {
    return {
        searchText: store.search.searchText,
        onBlur: ownProps.onBlur,
        onChange: ownProps.onChange,
        onFocus: ownProps.onFocus,
        onKeyPress: ownProps.onKeyPress
    };
}

function mapDispatchToProps(dispatch) {
    return bindActionCreators({
    },
    dispatch);
}

SearchBarInput.propTypes = propTypes;
SearchBarInput.defaultProps = defaultProps;

export default connect(mapStateToProps, mapDispatchToProps)(SearchBarInput);