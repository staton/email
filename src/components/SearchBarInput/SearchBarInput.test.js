import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow, mount } from 'enzyme';
import { SearchBarInput } from './SearchBarInput';

describe('<SearchBarInput />', () => {

    it('renders correctly', () => {
        const wrapper = shallow(<SearchBarInput />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('calls all events when triggered', () => {
        const onBlur = jest.fn();
        const onFocus = jest.fn();
        const onKeyPress = jest.fn();
        const wrapper = mount(
            <SearchBarInput 
                onBlur={onBlur}
                onFocus={onFocus}
                onKeyPress={onKeyPress}
            />
        );
        let input = wrapper.find('input');

        input.simulate('blur');
        expect(onBlur).toHaveBeenCalledTimes(1);

        input.simulate('focus');
        expect(onFocus).toHaveBeenCalledTimes(1);

        input.simulate('keypress');
        expect(onKeyPress).toHaveBeenCalledTimes(1);

        wrapper.unmount();
    });

    it('contains an input with a value equal to the searchText', () => {
        const searchText = "ABcd398352222";
        const wrapper = shallow(<SearchBarInput searchText={searchText} />);
        expect(wrapper.find('input').prop('value')).toEqual(searchText);
    });

});