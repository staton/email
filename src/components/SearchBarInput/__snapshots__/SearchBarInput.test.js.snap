// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBarInput /> renders correctly 1`] = `
<input
  className="SearchBarInput"
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  onKeyPress={[Function]}
  placeholder="Search..."
  type="text"
/>
`;
